client/data/settings/hooks.js

```diff
@@ -10,46 +10,61 @@ import { useSelect, useDispatch } from '@wordpress/data';
  */
 import { STORE_NAME } from '../constants';
 
-export const useEnabledPaymentMethodIds = () =>
-	useSelect( ( select ) => {
-		const { getEnabledPaymentMethodIds } = select( STORE_NAME );
-		const { updateEnabledPaymentMethodIds } = useDispatch( STORE_NAME );
```

Comment:
`useDispatch` was called within `useSelect` - see https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level